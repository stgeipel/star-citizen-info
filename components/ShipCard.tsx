import Shipstatus from './Shipstatus'
import { IShip } from '../types/Ship'

const ShipCard = ({ data: ship }: { data: IShip }) => (
  <div className="flex flex-col rounded-lg shadow-lg overflow-hidden">
    <div className="flex-1 bg-white p-6 flex flex-col justify-between">
      <div className="flex-1">
        <a href="#" className="text-base text-gray-500">
          {ship.company.name}
        </a>
        <div className="float-right">
          <Shipstatus status={ship.develop_status}></Shipstatus>
        </div>
        <h3 className="mt-2 text-xl leading-7 font-semibold text-gray-900">{ship.name}</h3>
        <p className="mt-3 text-base leading-6 text-gray-500">
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto accusantium
          praesentium eius, ut atque fuga culpa, similique sequi cum eos quis dolorum.
        </p>
      </div>
    </div>
  </div>
)

export default ShipCard
