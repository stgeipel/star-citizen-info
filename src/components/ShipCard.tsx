import React from 'react';
import Shipstatus from './Shipstatus';
import { Ship } from '../types/Ship';
import { formatNumber } from '../utils/fotmattings';

export interface Props {
  data: Ship;
}

const ShipCard: React.FC<Props> = ({ data: ship }) => (
  <div className="flex flex-col rounded-lg shadow-lg overflow-hidden">
    <div className="flex-1 dark:bg-gray-800 p-6 flex flex-col justify-between bg-white">
      <div className="flex-1">
        <a href="#" className="text-base text-gray-500">
          {ship.company.name}
        </a>
        <div className="float-right">
          <Shipstatus status={ship.ship_develop_status}></Shipstatus>
        </div>
        <h3 className="mt-2 text-xl leading-7 font-semibold text-gray-900 dark:text-white">
          {ship.name}
        </h3>
        <div className="mt-3 sm:mx-6">
          <dl className="grid grid-cols-1 col-gap-4 row-gap-8 sm:grid-cols-2">
            <div className="sm:col-span-1">
              <dt className="text-sm leading-5 font-medium text-gray-500">
                Cargokapatzit√§t
              </dt>
              <dd className="mt-1 text-sm leading-5 text-gray-900 dark:text-gray-300">
                {ship.cargo_capacity}
              </dd>
            </div>
            <div className="sm:col-span-1">
              <dt className="text-sm leading-5 font-medium text-gray-500 ">
                Preis
              </dt>
              <dd className="mt-1 text-sm leading-5 text-gray-900 dark:text-gray-300">
                {formatNumber(ship.price)} aUEC
              </dd>
            </div>
            {/* <div className="sm:col-span-1">
              <dt className="text-sm leading-5 font-medium text-gray-500">Email address</dt>
              <dd className="mt-1 text-sm leading-5 text-gray-900">margotfoster@example.com</dd>
            </div>
            <div className="sm:col-span-1">
              <dt className="text-sm leading-5 font-medium text-gray-500">Salary expectation</dt>
              <dd className="mt-1 text-sm leading-5 text-gray-900">$120,000</dd>
            </div> */}
          </dl>
        </div>
      </div>
    </div>
  </div>
);

export default ShipCard;
