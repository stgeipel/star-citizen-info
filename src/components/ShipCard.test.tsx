import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ShipCard from './ShipCard';

describe('ShipCard', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <ShipCard
        data={{
          name: 'Constellation Aquila',
          price: 10,
          cargo_capacity: 10,
          ship_develop_status: { value: 'Flight Ready' },
          classification: { shiptype: 'transport' },
          company: { name: 'Robert Space Industries', short_name: 'RSI' },
        }}
      ></ShipCard>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
