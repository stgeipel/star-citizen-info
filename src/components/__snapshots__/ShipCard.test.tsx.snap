// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShipCard matches snapshot 1`] = `
<div
  className="flex flex-col rounded-lg shadow-lg overflow-hidden"
>
  <div
    className="flex-1 dark:bg-gray-800 p-6 flex flex-col justify-between bg-white"
  >
    <div
      className="flex-1"
    >
      <a
        className="text-base text-gray-500"
        href="#"
      >
        Robert Space Industries
      </a>
      <div
        className="float-right"
      >
        <Shipstatus
          status={
            Object {
              "value": "Flight Ready",
            }
          }
        />
      </div>
      <h3
        className="mt-2 text-xl leading-7 font-semibold text-gray-900 dark:text-white"
      >
        Constellation Aquila
      </h3>
      <div
        className="mt-3 sm:mx-6"
      >
        <dl
          className="grid grid-cols-1 col-gap-4 row-gap-8 sm:grid-cols-2"
        >
          <div
            className="sm:col-span-1"
          >
            <dt
              className="text-sm leading-5 font-medium text-gray-500"
            >
              Cargokapatzit√§t
            </dt>
            <dd
              className="mt-1 text-sm leading-5 text-gray-900 dark:text-gray-300"
            >
              10
            </dd>
          </div>
          <div
            className="sm:col-span-1"
          >
            <dt
              className="text-sm leading-5 font-medium text-gray-500 "
            >
              Preis
            </dt>
            <dd
              className="mt-1 text-sm leading-5 text-gray-900 dark:text-gray-300"
            >
              10
               aUEC
            </dd>
          </div>
        </dl>
      </div>
    </div>
  </div>
</div>
`;
